src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -339,5 +340,121 @@ public void ValidateGraphQLReservedNaming_DatabaseColumns(string dbColumnName, s
                 actual: isViolationWithGraphQLGloballyDisabled,
                 message: "Unexpected failure. fieldName: " + dbColumnName + " | fieldMapping:" + mappedName);
         }
+
+        /// <summary>
+        /// Test to validate successful inference of relationship data based on data provided in the config and the metadata
+        /// collected from the MsSql database.
+        /// </summary>
+        [TestMethod, TestCategory(TestCategory.MSSQL)]
+        public async Task ValidateInferredRelationshipInfoForMsSql()
+        {
+            DatabaseEngine = TestCategory.MSSQL;
+            await SetupTestFixtureAndInferMetadata();
+            ValidateInferredRelationshipInfoForTables();
+        }
```

Comment:
This test has an issue with artifact config files. It always runs successfully the first time the test is ran or debugged. Any subsequent runs result in a "null reference exception" , and the console shows loading two different config files. Please check and make sure this test runs with a clean filesystem each time.

Reply:
For some reason the first run always uses the file from the test project. then uses the config from the "service" project for subsequent runs (perhaps it is left as an artifact). -You can repro if from the service project config, you remove the "stock" entity.
-while the repo doesn't have "stock" entity removed, this is still a bug where the wrong config is retreived for subsequent test runs.

Reply:
fixed by Ayush in earlier iteration